ethtxmanager/ethtxmanager.go

```diff
@@ -36,6 +36,41 @@ var (
 	ErrExecutionReverted = errors.New("execution reverted")
 )
 
+// nonceMap is a struct that represents a mapping of Ethereum addresses to nonces.
+type nonceMap struct {
+	m map[common.Address]uint64
+
+	lock sync.RWMutex
+}
+
+// newNonceMap creates a new instance of nonceMap.
+// It returns a pointer to the newly created nonceMap.
+func newNonceMap() *nonceMap {
+	return &nonceMap{
+		m: make(map[common.Address]uint64, 0),
+	}
+}
+
+// get retrieves the nonce value for the given address.
+// It returns the nonce value and a boolean indicating whether the nonce exists in the map.
+func (n *nonceMap) get(addr common.Address) (uint64, bool) {
+	n.lock.RLock()
+	defer n.lock.RUnlock()
```

Comment:
The logic is very straight forward, no needed to defer here, you can just: lock, retrieve, unlock

Reply:
Fix: https://github.com/0xPolygonHermez/zkevm-node/pull/3320/commits/f6e92fb3753524d74123597a818980499f29ec10